devito/dse/rewriters.py

```diff
@@ -127,7 +115,8 @@ def _extract_increments(self, cluster, template, **kwargs):
                     extracted = e.rhs
                 else:
                     extracted = e.rhs.func(*[i for i in e.rhs.args if i != e.lhs])
-                processed.extend([Eq(handle, extracted), e.func(e.lhs, handle)])
+                processed.extend([e.func(handle, extracted, is_Increment=False),
```

Comment:
shouldn't e.func know if it's an increment or ha `is_Increment=e.is_increment`?

Reply:
this first expression isn't an increment (it's just a pure assignment) hance the `is_Increment=False`